### PR TITLE
make payment.extra non-optional

### PR DESCRIPTION
**Pull request by prusnak**
this makes handling in tasks.py so much easier

### CODE REVIEW

File: lnbits/extensions/cashu/tasks.py [hunk 1]
```diff
@@ -28,6 +28,10 @@ async def wait_for_paid_invoices():
 
 
 async def on_invoice_paid(payment: Payment) -> None:
-    if payment.extra and not payment.extra.get("tag") == "cashu":
+    if not payment.extra:
```
Comment (prusnak) [inline lnbits/extensions/cashu/tasks.py:31]: This is where the logic was changed, not just a simple refactor. I think the suggested behaviour is more correct than the old one.

Comment (dni) [reply lnbits/extensions/cashu/tasks.py:31]: we could also try to evalute if there is a good reason for payment.extra to be Optional, maybe we can get rid of it altogether?

**Comment by prusnak** [comment]
Rebased to include changes in recently merged https://github.com/lnbits/lnbits/pull/1286 and https://github.com/lnbits/lnbits/pull/1281

**Comment by dni** [comment]
loving it, LGTM

**Review by dni — APPROVED** [review]
lgtm

**Comment by prusnak** [comment]
rebased to cover https://github.com/lnbits/lnbits/pull/1293

**Comment by prusnak** [comment]
if these are merged earlier than this PR, this PR would need to be reworked to address these as well: #1295 #1296 #1297 #1299

**Pull request merged**